python/pdo/submitter/ccf/ccf_submitter.py

```diff
@@ -92,18 +68,21 @@ def submit_read_request(self, tx_method, tx_params) :
         for wait in self.read_backoff_duration :
             time.sleep(wait)
 
-            #if read fails due to lack of global commit, result will be None
-            response = self.submit_rpc(tx_method, tx_params)
-            if response.result is not None :
-                return response.result
-
+            try:
+                #if read fails due to lack of global commit, response.status_code will not be OK
+                response = self.submit_rpc(tx_method, tx_params)
+                if response.status_code == http.HTTPStatus.OK:
+                    return response.body.json()
+            except Exception as e: 
+                raise
         #read failed even after retries
-        raise Exception(response.error)
+        raise Exception("read request failed after multiple retries for tx_method {}".format(str(tx_method)))
 
```

Comment:
any reason why you are dropping response.error from the exception? seems like useful information

Reply:
i guess after the version change, response did not have an error field. may be there is a replacement, i haven't checked.